src/patchwork/components/PatchworkDocEditor.tsx

```diff
@@ -309,8 +305,17 @@ export const PatchworkDocEditor: React.FC<{
 
   const activeDoc = selectedBranchUrl ? branchDoc : doc;
   const activeChangeDoc = selectedBranchUrl ? changeBranchDoc : changeDoc;
-
-  const annotations = useAnnotations({
+  const activeHandle = selectedBranchUrl ? branchHandle : handle;
+
+  const {
+    annotations,
+    annotationGroups,
+    selectedAnchors,
+    setHoveredAnchor,
+    setSelectedAnchors,
+    setHoveredAnnotationGroupId,
+    setSelectedAnnotationGroupId,
+  } = useAnnotations({
```

Comment:
I find it helpful to consider the selection/hover state of anchors and group annotations separately because their behavior is different. For example, if you hover over an anchor, it highlights all annotation groups in which it is contained together with the other anchors in these annotation groups. In contrast, if you hover over an annotation group, it only highlights the anchors in that annotation group.